src/routes_admin.py

```diff
@@ -288,7 +322,12 @@ def staff_add_extension(netid, cid, aid):
                 return util.error("Start must be before End.")
 
             for student_netid in student_netids:
-                db.add_extension(cid, aid, student_netid, max_runs, start, end)
+                end_str = util.parse_form_datetime(request.form["end"]).strftime("%Y-%m-%dT%H:%M")
+                # avoid that weird race condition - start run 5 min after, but with a container due date of the original time
+                ext_end = (util.parse_form_datetime(request.form["end"]) + timedelta(minutes=5)).strftime("%Y-%m-%dT%H:%M")
+                run_id = db.generate_new_id()
+                print("Schedule result:", add_or_edit_scheduled_run(cid, aid, run_id, {"run_time": ext_end, "due_time": end_str, "name": f"Extension Run - {student_netid}", "config": request.form['config'], "roster": student_netid}, None), flush=True)
+                db.add_extension(cid, aid, student_netid, max_runs, start, end, run_id)
```

Comment:
Not really related to the PR, but ideally these non-atomic writes to the DB should either be made atomic or have some failure-checking to display errors for students whose extensions weren't updated.